plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/EditConnectionHandler.java

```diff
@@ -28,11 +30,16 @@
 
 	@Override
 	public Object execute(ExecutionEvent event) throws ExecutionException {
-		IConnection connection = UIUtils.getFirstElement(HandlerUtil.getCurrentSelection(event), IConnection.class);
-		return openConnectionWizard(connection, event);
+	    ConnectionWrapper connectionWrapper = UIUtils.getFirstElement(HandlerUtil.getCurrentSelection(event), ConnectionWrapper.class);
```

Comment:
We here depend on a v3 specific class ConnectionWrapper. This will thus only work in v3 (we dont have v2 any more anyhow). Moving this from openshift.common.ui to openshift.ui  makes perfect sense. But I think that we therefore should also correct the activation expression from testing the common IConnection to the v3 ConnectionWrapper (openshift.ui/plugin.xml, l1485):
```
<adapt type="org.jboss.tools.openshift.common.core.connection.IConnection" />
```